src/backend/utils.ts

```diff
@@ -1288,94 +1290,81 @@ export interface ProgressCallback {
  */
 export async function downloadFile(
   url: string,
-  dest: string,
+  directory: string,
+  fileName: string,
   abortController: AbortController,
-  progressCallback?: ProgressCallback
-): Promise<void> {
+  progressCallback?: ProgressCallback,
+  onCompleted?: (file: File) => void,
+  onCancel?: (item: DownloadItem) => void
+): Promise<DownloadItem> {
```

Comment:
we return `DownloadItem` so that we can handle pause, resume, etc. in the calling code (hp store)